x/ibc/03-connection/types/version.go

```diff
@@ -1,96 +1,45 @@
-// of compatible versions for the connection.
-func PickVersion(counterpartyVersions, compatibleVersions []string) string {
-
-	n := len(counterpartyVersions)
-	m := len(compatibleVersions)
-
-	// aux hash maps to lookup already seen versions
-	counterpartyVerLookup := make(map[string]bool)
-	compatibleVerLookup := make(map[string]bool)
-
-	// versions suported
-	var supportedVersions []string
-
-	switch {
-	case n == 0 || m == 0:
-		return ""
-	case n == m:
-		for i := n - 1; i >= 0; i-- {
-			counterpartyVerLookup[counterpartyVersions[i]] = true
-			compatibleVerLookup[compatibleVersions[i]] = true
-
-			// check if we've seen any of the versions
-			if _, ok := compatibleVerLookup[counterpartyVersions[i]]; ok {
-				supportedVersions = append(supportedVersions, counterpartyVersions[i])
-			}
+import (
+	"strings"
 
-			if _, ok := counterpartyVerLookup[compatibleVersions[i]]; ok {
-				// TODO: check if the version is already in the array
-				supportedVersions = append(supportedVersions, compatibleVersions[i])
-			}
-		}
-	case n > m:
-		for i := n - 1; i >= m; i-- {
-			counterpartyVerLookup[compatibleVersions[i]] = true
-		}
-
-		for i := m - 1; i >= 0; i-- {
-			counterpartyVerLookup[counterpartyVersions[i]] = true
-			compatibleVerLookup[compatibleVersions[i]] = true
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
 
-			// check if we've seen any of the versions
-			if _, ok := compatibleVerLookup[counterpartyVersions[i]]; ok {
-				supportedVersions = append(supportedVersions, counterpartyVersions[i])
-			}
+var (
+	// DefaultIBCVersion represents the latest supported version of IBC
+	DefaultIBCVersion = "1.0.0"
+)
 
-			if _, ok := counterpartyVerLookup[compatibleVersions[i]]; ok {
-				supportedVersions = append(supportedVersions, compatibleVersions[i])
-			}
-		}
+// GetCompatibleVersions returns a descending ordered set of compatible IBC versions
+// for the caller chain's connection end.
+func GetCompatibleVersions() []string {
+	return []string{DefaultIBCVersion}
+}
 
-	case n < m:
-		for i := m - 1; i >= n; i-- {
-			compatibleVerLookup[compatibleVersions[i]] = true
+// IsSupportedVersion returns true if the version provided is supported.
+func IsSupportedVersion(version string, supportedVersions []string) bool {
+	for _, supportedVer := range supportedVersions {
+		if supportedVer == strings.TrimSpace(version) {
+			return true
 		}
+	}
+	return false
+}
 
-		for i := n - 1; i >= 0; i-- {
-			counterpartyVerLookup[counterpartyVersions[i]] = true
-			compatibleVerLookup[compatibleVersions[i]] = true
-
-			// check if we've seen any of the versions
-			if _, ok := compatibleVerLookup[counterpartyVersions[i]]; ok {
-				supportedVersions = append(supportedVersions, counterpartyVersions[i])
-			}
+// PickVersion iterates over the descending ordered set of compatible IBC versions
+// and selects the first version that is supported by the counterparty.
```

Comment:
I think we should pick the **latest** version that is supported by the counterparty by default (i.e sort desc the compatible versions and pick the first one).

Reply:
We could also have in the future an additional `PickVersion<Major/Minor>` that chooses any version from the same major or minor release respectively.

Reply:
> I think we should pick the **latest** version that is supported by the counterparty by default (i.e sort desc the compatible versions and pick the first one).

this should be the current behavior, the `GetCompatibleVersions` is expected to return in order from latest and descend to oldest

Reply:
Is this documented somewhere? If so, it should be

Reply:
```
// GetCompatibleVersions returns a descending ordered set of compatible IBC versions
// for the caller chain's connection end.
```

is there somewhere else I should note this?